cranelift/codegen/src/ir/entities.rs

```diff
@@ -394,6 +394,32 @@ impl Heap {
     }
 }
 
+/// An opaque reference to a [heap](https://en.wikipedia.org/wiki/Memory_management#DYNAMIC).
```

Comment:
This is more than just a reference to a heap, right? It's (i) the heap reference, (ii) the access flags, and (iii) a fixed offset. Basically, everything needed for a heap load/store that is not an SSA value, if I understand right. Could we note that here?

Also let's note somewhere that this is a little different than other entities in CLIF: it isn't an explicitly-defined entity in the preamble (like `heap0 = ...` or `gv0 = ...`) but rather is logically a piece of an instruction (of `HeapLoad` or `HeapStore` format) and one is nominally created for each use-site (each instruction).

Reply:
Ah this is just a copy-paste of `ir::Heap`'s docs that I forgot to update.